source/synthDrivers/oneCore.py

```diff
@@ -221,24 +223,45 @@ def _callback(self, bytes, len, markers):
 			log.debug("Done pushing audio")
 		self._processQueue()
 
-	def _getAvailableVoices(self, onlyValid=True):
+	def _getVoiceInfoFromOnecoreVoiceString(self, voiceStr):
+		"""
+		Produces an NVDA VoiceInfo object representing the given voice string from Onecore speech.
+		"""
+		# The voice string is made up of the ID, the language, and the display name.
+		ID,language,name=voiceStr.split(':')
+		language=language.replace('-','_')
+		return VoiceInfo(ID,name,language=language)
+
+	def _getAvailableVoices(self):

```

Comment:
I dont think this will be cached by default, is that a problem? It seems like this might be a slow operation.

Reply:
It is very slow, yes. However, the SynthDriver base class has an availableVoices property that by default calls _getAvailableVoices. This property is cached. As long as we never call _getAvailableVoices we are okay.